documentation/src/main/asciidoc/backend-lucene.asciidoc

```diff
@@ -14,14 +12,184 @@ but the defaults might not suit everyone.
+it will instead be indexed in directory
+`/path/to/my/root/orders/`.
+
+===== Filesystem access strategy
+
+The default strategy for accessing the filesystem is determined automatically
+based on the operating system and architecture.
+It should work well in most situations.
+
+For situations where a different filesystem access strategy is needed,
+Hibernate Search exposes a configuration property at the backend level:
+
+[source]
+----
+hibernate.search.backends.<backend-name>.directory.filesystem_access.strategy = auto (default)
+----
+
+Allowed values are:
+
+* `auto` (default): lets Lucene select the most appropriate implementation
+based on the operating system and architecture.
+* `simple`: a straightforward strategy based on `Files.newByteChannel`.
+See `org.apache.lucene.store.SimpleFSDirectory`.
+* `mmap`: uses `mmap` for reading, and `FSDirectory.FSIndexOutput` for writing.
+See `org.apache.lucene.store.MMapDirectory`.
+* `nio`: uses ``java.nio.channels.FileChannel``'s positional read for concurrent reading,
+and `FSDirectory.FSIndexOutput` for writing.
+See `org.apache.lucene.store.NIOFSDirectory`.
+
+[NOTE]
+====
+Make sure to refer to Javadocs of these `Directory`
+implementations before changing this setting.
+Implementations offering better performance
+also bring issues of their own.
+====
+
+===== Other configuration options
+
+The `local-filesystem` directory also allows configuring a
+<<backend-lucene-configuration-directory-locking-strategy,locking strategy>>.
+
+[[backend-lucene-configuration-directory-local-heap]]
+==== Local heap storage
+
+The `local-heap` directory type will store indexes in the local JVM's heap.
+
+As a result, indexes contained in a `local-heap` directory are *lost when the JVM shuts down*.
+
+This directory type is only provided for use in *testing configurations*
+with *small indexes* and *low concurrency*,
+where it could slightly improve performance.
+In setups requiring larger indexes and/or high concurrency,
+a <<backend-lucene-configuration-directory-local-filesystem,filesystem-based directory>>
+will achieve better performance.
+
+The `local-heap` directory does not offer any specific option
+beyond the <<backend-lucene-configuration-directory-locking-strategy,locking strategy>>.
+
+[[backend-lucene-configuration-directory-locking-strategy]]
+==== Locking strategy
+// Search 5 anchors backward compatibility
+[[search-configuration-directory-lockfactories]]
+
+In order to write to an index, Lucene needs to acquire a lock to ensure no other application instance
+writes to the same index concurrently.
+Each directory type comes with a default locking strategy that should work well enough in most situations.
+
+For those (very) rare situations where a different locking strategy is needed,
+Hibernate Search exposes a configuration property at the backend level:
+
+[source]
+----
+hibernate.search.backends.<backend-name>.directory.locking.strategy = /path/to/my/root
```

Comment:
Does `/path/to/my/root` make sense here?

Reply:
It was a copy/paste error. Fixed.